sdk/core/azure-core/azure/core/pipeline/_base.py

```diff
@@ -74,7 +75,7 @@ def send(self, request):
                 raise
         else:
             _await_result(self._policy.on_response, request, response)
-        return response
+        return response # type: ignore
```

Comment:
mypy is right to warn about this though. The annotation is wrong: this method now returns `Optional[PipelineResponse]`

Reply:
Yes. But unfortunately, changing return type from PipelineResponse to Optional[PipelineResponse] is a breaking change.

It will break mypy for its consumers.

Reply:
The code change itself is breaking. Preventing mypy from warning people about it doesn't help.

Reply:
The behavior change is from raise -> return None. This only breaks non-working code. (in 99% of the case)

mypy breaking will be more impactful and breaks working code.

Reply:
It doesn't make that non-working code work though. It just replaces an UnboundLocalError in this method with an AttributeError somewhere else because none of our code expects Pipeline.run() to return None. Broken code stays broken while working code gains a new failure mode hidden from mypy.